pilot/pkg/model/context.go

```diff
@@ -99,12 +99,15 @@ func (e *Environment) AddMetric(metric monitoring.Metric, key string, proxy *Pro
 	}
 }
 
-// Generator creates the response for a typeURL DiscoveryRequest. If no generator is associated
+// Request is an alias for array of marshalled resources.
+type Response = []*any.Any
+
+// ResourceGenerator creates the response for a typeURL DiscoveryRequest. If no generator is associated
 // with a Proxy, the default (a networking.core.ConfigGenerator instance) will be used.
 // The server may associate a different generator based on client metadata. Different
 // WatchedResources may use same or different Generator.
-type Generator interface {
-	Generate(node *Proxy, push *PushContext, w *WatchedResource) []*any.Any
+type ResourceGenerator interface {
```

Comment:
I felt Generator is too generic, may be we should call `XdsResourceGenerator` ? I ended up with `ResourceGenerator`

Reply:
XdsResourceGenerator?

Reply:
makes sense. Let me rename